### ISSUE TITLE
[bug] impossible to enable C++20

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#6856]
I was trying to enable C++20 in Conan, however, I am getting an error:
```
ERROR: The specified 'compiler.cppstd=20' is not available for 'apple-clang 11.0'. Possible values are ['11', '14', '17', '98', 'gnu11', 'gnu14', 'gnu17', 'gnu98']'
```

I have `20` in my **settings.yml**:
```
63     apple-clang: &apple_clang
64         cppstd: [None, 98, 20, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20, 20]`
```

this is incorrect, as the compiler has support for C++20.

/cc @jgsogo 

### Environment Details (include every applicable attribute)
  * Operating System+version: macOS 10.15.4
  * Compiler+version: Apple clang version 11.0.3 (clang-1103.0.32.29)
  * Conan version: Conan 1.24.0
  * Python version: Python 2.7.16

### Steps to reproduce (Include if Applicable)

`conan create . boost/1.72.0 -k -s compiler.cppstd=20`

(within conan-center-index repository)

**Comment by memsharded** [comment]
The ``_cppstd_apple_clang`` is incomplete.

Which is the first apple-clang version to support c++20? Should it be ``-std=c++2a`` I guess, because support is not complete yet.

**Comment by SSE4** [comment]
Clang and Apple Clang don't have **full** support of C++20 (yet).
the error message is weird, why does it list just:
```
Possible values are ['11', '14', '17', '98', 'gnu11', 'gnu14', 'gnu17', 'gnu98']'
```
while my settings.yml definetely has **20**?

**Comment by memsharded** [comment]
Because is the ``settings_preprocessor`` the one that is pre-validating the ``cppstd`` based on compiler versions, relying on the ``_cppstd_apple_clang`` function.

**Comment by memsharded** [comment]
Something like this might fix it:

```diff
diff --git a/conans/client/build/cppstd_flags.py b/conans/client/build/cppstd_flags.py
index b209bd0ad..32933b4d6 100644
--- a/conans/client/build/cppstd_flags.py
+++ b/conans/client/build/cppstd_flags.py
@@ -118,6 +118,16 @@ def _cppstd_apple_clang(clang_version, cppstd):
         v17 = "c++17"
         vgnu17 = "gnu++17"

+    if Version(clang_version) >= "10.0":
+        # Not confirmed that it didn't work before 9.1 but 1z is still valid, so we are ok
+        v20 = "c++2a"
+        vgnu20 = "gnu++2a"
+
+    if Version(clang_version) >= "11.0":
+        # Not confirmed that it didn't work before 9.1 but 1z is still valid, so we are ok
+        v20 = "c++20"
+        vgnu20 = "gnu++20"
+
     flag = {"98": v98, "gnu98": vgnu98,
             "11": v11, "gnu11": vgnu11,
             "14": v14, "gnu14": vgnu14,
```

But need to confirm the exact versions. Please @SSE4 send a PR with a fix and tests when possible (for 1.25)
